File: VoiceRecorder.Audio/AutoTuneSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VoiceRecorder.Audio
{
    public class AutoTuneSettings
    {
        public AutoTuneSettings()
        {
            // set up defaults
            SnapMode = true;
            PluggedIn = true;
            AutoPitches = new HashSet<Note>();
            AutoPitches.Add(Note.C);
            AutoPitches.Add(Note.CSharp); 
            AutoPitches.Add(Note.D);
            AutoPitches.Add(Note.DSharp);
            AutoPitches.Add(Note.E);
            AutoPitches.Add(Note.F);
            AutoPitches.Add(Note.FSharp);
            AutoPitches.Add(Note.G);
            AutoPitches.Add(Note.GSharp); 
            AutoPitches.Add(Note.A);
            AutoPitches.Add(Note.ASharp); 
            AutoPitches.Add(Note.B);
            VibratoDepth = 0.0;
            VibratoRate = 4.0;
            AttackTimeMilliseconds = 0.0;
        }

        public bool Enabled { get; set; }
        public bool SnapMode { get; set; } // snap mode finds a note from the list to snap to, non-snap mode is provided with target pitches from outside
        public double AttackTimeMilliseconds { get; set; }
        public HashSet<Note> AutoPitches { get; private set; }
        public bool PluggedIn { get; set; } // not currently used
        public double VibratoRate { get; set; } // not currently used
        public double VibratoDepth { get; set; } 

        /*
         *  vibRateSlider = new GuiSlider(0.2, 20.0, 4.0);
            vibDepthSlider = new GuiSlider(0.0, 0.05, 0);
            attackSlider = new GuiSlider(0.0, 200, 0.0);
         */
    }

    public enum Note
    {
        C, CSharp, D, DSharp, E, F, FSharp, G, GSharp, A, ASharp, B
    }
}

File: ~VB Port/VoiceRecorder.Audio/AutoTuneSettings.vb
Imports System.Text

Namespace VoiceRecorder.Audio
    Public Class AutoTuneSettings
        Public Sub New()
            ' set up defaults
            SnapMode = True
            PluggedIn = True
            AutoPitches = New HashSet(Of Note)
            AutoPitches.Add(Note.C)
            AutoPitches.Add(Note.CSharp)
            AutoPitches.Add(Note.D)
            AutoPitches.Add(Note.DSharp)
            AutoPitches.Add(Note.E)
            AutoPitches.Add(Note.F)
            AutoPitches.Add(Note.FSharp)
            AutoPitches.Add(Note.G)
            AutoPitches.Add(Note.GSharp)
            AutoPitches.Add(Note.A)
            AutoPitches.Add(Note.ASharp)
            AutoPitches.Add(Note.B)
            VibratoDepth = 0.0
            VibratoRate = 4.0
            AttackTimeMilliseconds = 0.0
        End Sub

        Public Property Enabled As Boolean

        Public Property SnapMode As Boolean

        Public Property AttackTimeMilliseconds As Double

        Private _autoPitches As HashSet(Of Note)
        Public Property AutoPitches As HashSet(Of Note)
            Get
                Return _autoPitches
            End Get
            Private Set(ByVal value As HashSet(Of Note))
                _autoPitches = value
            End Set
        End Property

        Public Property PluggedIn As Boolean

        Public Property VibratoRate As Double

        Public Property VibratoDepth As Double

        '        
        '         *  vibRateSlider = New GuiSlider(0.2, 20.0, 4.0)
        '            vibDepthSlider = New GuiSlider(0.0, 0.05, 0)
        '            attackSlider = New GuiSlider(0.0, 200, 0.0)
        '        
    End Class


    Public Enum Note
        C
        CSharp
        D
        DSharp
        E
        F
        FSharp
        G
        GSharp
        A
        ASharp
        B
    End Enum
End Namespace
